nx/guides/advanced/backend_comparison.livemd

```diff
@@ -0,0 +1,267 @@
+x = Nx.tensor([1.0, 2.0, 3.0, 4.0])
+y = Nx.tensor([0.5, 1.5, 2.5, 3.5])
+
+IO.puts("Input tensors:")
+IO.inspect(x, label: "x")
+IO.inspect(y, label: "y")
+```
+
+## Preparing the function for comparing
+
+In order to ensure the same `id` for each node in the graph while our function traverses it on both backends, we need to use `Nx.Defn.debug_expr/1` to pre-compile `SimpleComputation.compute/2`.
+
+This is a trick to make sure the same expression is passed on both `Nx.Defn.jit/2` calls and should not be used liberally.
+
+```elixir
+expr = Nx.Defn.debug_expr(&SimpleComputation.compute/2).(x, y)
+
+precompiled = fn _x, _y -> expr end
+```
+
+## Running with Backend A
+
+Let's run our computation with the first backend (BinaryBackend in this example, but could be any backend):
+
+```elixir
+# Clean up and create output directory
+File.rm_rf!("/tmp/backend_a")
+File.mkdir_p!("/tmp/backend_a")
+
+# Run computation with debug output enabled
+result_a = Nx.Defn.jit(
+  precompiled,
+  compiler: Nx.Defn.Evaluator,
+  debug_options: [save_path: "/tmp/backend_a"]
+).(x, y)
+
+IO.puts("\n✅ Backend A completed")
+IO.inspect(result_a, label: "Result A")
+IO.puts("Backend: #{inspect(result_a.data.__struct__)}")
+
+# Show what files were generated
+files_a = File.ls!("/tmp/backend_a")
+IO.puts("\nGenerated #{length(files_a)} node files:")
+Enum.each(files_a, &IO.puts("  - #{&1}"))
+```
+
+## Examining the Output Files
+
+Let's look at what the `.exs` files contain:
+
+```elixir
+# Read and display one of the generated files
+example_file = File.ls!("/tmp/backend_a") |> List.last()
+content = File.read!(Path.join("/tmp/backend_a", example_file))
+
+IO.puts("=== Content of #{example_file} ===\n")
+IO.puts(content)
+```
+
+Notice the format:
+
+* **Node ID** - Unique identifier for this computation node
+* **Operation** - The operation being performed (e.g., `:add`, `:multiply`, `:parameter`)
+* **Arguments** - List containing parameters and tensors as strings
+* **Result** - Executable code that reconstructs the output tensor from binary
+
+## Verifying Executability
+
+The result line is actual Elixir code that can be executed:
+
+```elixir
+# Extract and execute the result line
+result_line = content
+  |> String.split("\n")
+  |> Enum.find(&String.starts_with?(&1, "result = "))
+  |> String.replace("result = ", "")
+
+# Execute it to reconstruct the tensor
+{reconstructed, _} = Code.eval_string(result_line)
+
+IO.puts("✅ Successfully reconstructed tensor from .exs file!")
+IO.inspect(reconstructed, label: "Reconstructed tensor")
+IO.puts("\nProperties:")
+IO.puts("  Type: #{inspect(reconstructed.type)}")
+IO.puts("  Shape: #{inspect(reconstructed.shape)}")
+IO.puts("  Backend: #{inspect(reconstructed.data.__struct__)}")
+```
+
+## Running with Backend B
+
+Now let's run the same computation with the swapped operations. We leave `Nx` on its default backend, but temporarily enable the Mimic stubs so the evaluator will capture the modified behaviour.
```

Comment:
```suggestion
Now let's run the same computation with the swapped operations. We leave `Nx` on its default backend, but temporarily enable the Mimic stubs so the evaluator will capture the modified behaviour.

In practice, this would be where we call a totally different backend to compare against the reference.
```